kafka-rest/src/main/java/io/confluent/kafkarest/controllers/RecordSerializerFacade.java

```diff
@@ -45,8 +47,15 @@
       Optional<RegisteredSchema> schema,
       JsonNode data,
       boolean isKey) {
+
     if (format.requiresSchema()) {
-      return schemaRecordSerializer.get().serialize(format, topicName, schema, data, isKey);
+      SchemaRecordSerializer schemaRecordSerializer;
+      try {
+        schemaRecordSerializer = schemaRecordSerializerProvider.get();
+      } catch (MultiException ise) {
```

Comment:
We should not rely on the DI framework's exception-on-failed-create for business logic. Instead, you should:

Option 1:
1. Make `SchemaRecordSerializer` an interface with current class renamed `SchemaRecrodSerializerImpl`.
2. Create a `ThrowingSchemaRecordSerializer`
3. In the `ControllersModule`, bind `SchemaRecordSerializer` to either `SchemaRecrodSerializerImpl` or `ThrowingSchemaRecordSerializer` based on whether schema registry is configured.

Option 2:
1. In `ControllersModule`, bind `Optional<SchemaRecordSerializer>` instead of `SchemaRecordSerializer`, to either an instance or empty based on whether schema registry is configured or not.
2. Inject `Optional<SchemaRecordSerializer>`

I have preference for Option 1, but author's choice.